src/string/src/lib.rs

```diff
@@ -156,25 +156,17 @@ pub unsafe extern "C" fn strncat(s1: *mut c_char, s2: *const c_char, n: usize) -
 
 #[no_mangle]
 pub unsafe extern "C" fn strncmp(s1: *const c_char, s2: *const c_char, n: usize) -> c_int {
-    let s1 = platform::c_str_n(s1, n);
-    let s2 = platform::c_str_n(s2, n);
-
-    let min_len = n.min(s1.len()).min(s2.len());
-    for i in 0..min_len {
-        let val = s1[i] - s2[i];
-        if val != 0 {
-            return val as c_int;
+    let s1 = core::slice::from_raw_parts(s1 as *const c_uchar, n);
+    let s2 = core::slice::from_raw_parts(s2 as *const c_uchar, n);
```

Comment:
If `s2` points to a buffer of less than `n` bytes, the loop below can iterate beyond the end of the buffer. It would be highly unlikely, but I think it is possible. Particularly if `s2` points to an buffer without a null terminating character. Not sure if that is valid tho.

Reply:
If `val == 0` and `a != 0` are both true, `b != 0`.

Reply:
Basically the only situation in which that could happen (AFAICT) is when the user provides invalid strings and an invalid `n`, in which case all bets are off anyway.

Reply:
yeah, hence the comment above about the buffer pointed to by `s2` being smaller than `n` and not having a null terminating character. As I mentioned above, I'm not sure if that is a valid scenario. If it is not a valid, this can definitely be ignored.

Reply:
Yeah, makes sense